prqlc/prqlc/Cargo.toml

```diff
@@ -5,12 +5,9 @@ name = "prqlc"
 edition.workspace = true
 license.workspace = true
 repository.workspace = true
-rust-version.workspace = true
 version.workspace = true
 
-# Required for `cargo-msrv`, which doesn't yet support workspaces
-metadata.msrv = "1.73.0"
-
+rust-version = "1.73.0"
```

Comment:
I am not sure if this is the ideal situation as I believe the MSRV is modified by the cli feature.
(All downstream crates that don't use the cli feature would also be affected by this)

Reply:
In my opinion, it is not a good idea to manage the compiler itself and the CLI in one crate (much less make it a default feature).

Reply:
On the narrow question of whether the `rust-version` here is correct — it's fine if the version constraint could be a bit lower. We currently test that 1.73.0 covers all features. So I think we're good?

Reply:
My point is that it is the `cli` feature that pushes the MSRV to 1.73, if excluding the feature, MSRV is 1.70.
However, there is no way to express that in the current Cargo.toml, so the `rust-version` here must be 1.73 and downstream crates cannot use Rust 1.70.

This is just the current situation, but my point is that it may be difficult to combine those that should keep MSRV as low as possible for downstream considerations and those that should not into a single crate and manage them with feature flags.
I'm fine with it as it is for now, but ideally the CLI should be separated into a separate crate like `prqlc-cli`.

Reply:
Yes, I agree that's a downside with having them integrated. I do think there are benefits to having them in a single crate — in particular `cargo install prqlc`, but we have gone back & forth, there are tradeoffs...

Reply:
> I do think there are benefits to having them in a single crate — in particular `cargo install prqlc`, but we have gone back & forth, there are tradeoffs...

I do not see how there is any problem with `cargo install prqlc` changing to `cargo install prqlc-cli`.
Or rename other than CLI to something like `prqlc-core`.

In any case, this is a separate issue that needs to be taken up.

Reply:
#4946